### PR TITLE
Add brand-icon for collapsed navigation

### PR DESCRIPTION
**Pull request by pxlrbt**
Allows a brand icon for the collapsed sidebar. Default is 2 initials of the brand name;

https://user-images.githubusercontent.com/22632550/159794380-73f37e43-f28b-4548-b215-db5778468448.mov

By overwriting the `brand-icon.blade.php` you could add nice SVG icons:

https://user-images.githubusercontent.com/22632550/159794392-e6176421-ad79-4dec-9b09-9b954996e1ab.mov

### CODE REVIEW

**Comment by danharrin** [comment]
We should get some simple docs in [this section](https://filamentphp.com/docs/2.x/admin/appearance#changing-the-brand-logo) about how to switch between versions.

**Comment by pxlrbt** [comment]
Not sure what you mean by "how to switch between versions."
But I provided some info on how to set the brand icon.

Not sure whether it should be `components/brand-icon.blade.php` or `components/layouts/app/sidebar/brand-icon.blade.php` as it's only used in the sidebar header so far. What do you think?

File: packages/admin/resources/views/components/layouts/app/sidebar/index.blade.php [hunk 1]
```diff
@@ -30,6 +29,21 @@
         >
             <x-filament::brand />
         </a>
+
+        @if (config('filament.layout.sidebar.is_collapsible_on_desktop'))
+            <a
+                class="block w-full text-center"
+                href="{{ config('filament.home_url') }}"
+                @if (config('filament.layout.sidebar.is_collapsible_on_desktop'))
```
Comment (danharrin) [inline packages/admin/resources/views/components/layouts/app/sidebar/index.blade.php:37]: This is always true here

**Comment by danharrin** [comment]
> Not sure whether it should be `components/brand-icon.blade.php` or `components/layouts/app/sidebar/brand-icon.blade.php` as it's only used in the sidebar header so far. What do you think?

`components/brand-icon.blade.php` is good, I think. Then we can use it elsewhere.

File: packages/admin/resources/views/components/layouts/app/sidebar/index.blade.php (hunk 1 above)
Comment (pxlrbt) [reply packages/admin/resources/views/components/layouts/app/sidebar/index.blade.php]: Oops. You're right. Time for some sleep

**Comment by danharrin** [comment]
Stellar as always. Thank you.

**Comment by asefsoft** [comment]
Hi
`str` helper function just added to Laravel 9 and it is not available on older versions of Laravel. so using it here will break compatibility of filamnet with Laravel 8 and ...

Please use another methods or just use Str facade.

I just updated filament and faced with this exception on all pages:
`Call to undefined function str() (View: ..\vendor\filament\filament\resources\views\components\brand-icon.blade.php`

**Comment by pxlrbt** [comment]
@asefsoft Thanks. Already got a report on Discord and opened a PR with a fix: https://github.com/laravel-filament/filament/pull/1990

**Pull request merged**